### PR TITLE
Improve performance of LRNGrad on CPU builds

### PR DESCRIPTION
**Pull request by markdryan**
This PR improves the performance of the CPU version of LRNGrad by moving a number of computations out of its innermost loop.  The extent of the resulting performance gains is dependent on the depth_radius, the compiler and the number of CPUs available.  The BM_LRNGrad benchmark is showing a 1.48x performance gain for a depth_radius of 4 and a 1.31x gain for a depth_radius of 2 when built with gcc 7.3.0 and run on a single core machine.  With gcc 5.4.0 the results on a single core machine are less impressive but are nonetheless significant;  1.13x for a depth radius of 4 and 1.10x with a depth radius of 2. With gcc 7.3.0 on a 16 core machine speed ups of 1.4x and 1.2x can be observed using a depth radius of 4 and 2 respectively.

The PR also includes a new benchmark for LRNGradOp, used to measure the performance gains, and fixes some formatting issues with the lrn_op.cc file.

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @rryan: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

File: tensorflow/core/kernels/lrn_op.cc [hunk 1]
```diff
@@ -358,13 +360,15 @@ struct LaunchLRNGrad<CPUDevice, T> {
           }
           norm = alpha_ * norm + bias_;
           DCHECK_GT(norm, T(1e-6));
+          T pre_computed_pow = Eigen::numext::pow(norm, -beta_);
+          T activations_div_norm = alpha_beta_2_ * activations(i, j) / norm;
+          T gs = grads_shaped(i, j);
           for (int64 k = depth_begin; k < depth_end; ++k) {
-            T dyi = T(-2) * alpha_ * beta_ * in_shaped(i, k) *
-                    activations(i, j) / norm;
+            T dyi = in_shaped(i, k) * activations_div_norm;
```
Comment (rryan) [inline tensorflow/core/kernels/lrn_op.cc]: My only concern is that moving the division before the in_shaped * activations could result in a loss of precision. Is hoisting the division out of the loop crucial to the speedups you see?

**Comment by rryan** [comment]
Thank you @markdryan and sorry for the delay.

@ymodak I'm happy to review, but I don't have the ability to approve PRs.

**Comment by rryan** [comment]
@markdryan was clang-format required? I'm surprised there were differences, since all internal changes to TensorFlow require clang-format to be applied. I just ran clang-format on lrn_op.cc and there were no changes, so I suspect you're either using a different version or it didn't end up using our [.clang-format](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/.clang-format) file.

**Comment by gbaned** [comment]
@rryan Could you PTAL and approve.

File: tensorflow/core/kernels/lrn_op.cc (hunk 1 above)
Comment (glima) [reply tensorflow/core/kernels/lrn_op.cc]: I have run CIFAR-10 training on TensorFlow with/without this patch, for a large number of steps, and the accuracy on cifar10_eval afterward was pretty much equal. BTW, I have a follow-up patch on this being finalized, with a vectorized preamble for LRNGrad (this such hot path on training) that also keeps accuracy and brings good speed gains, to be proposed soon.

**Review by rryan — APPROVED** [review]
Looks great, thank you @markdryan!

**Pull request merged**